snapcraft/tests/fixture_setup.py

```diff
@@ -534,40 +534,46 @@ def _setUp(self):
+                    [{"name": "$NAME",
+                      "status": "$STATUS",
+                      "devices": $DEVICES}]
+                    ''').substitute({
+                        # Container name without remote prefix
+                        'NAME': self.name.split(':')[-1],
+                        'STATUS': self.status,
+                        'DEVICES': self.devices,
+                    }).encode('utf-8')
+            return '[]'.encode('utf-8')
+        elif (args[0][0] == 'lxc' and
+              args[0][1] in ['init', 'start', 'launch', 'stop']):
+            return self._lxc_create_start_stop(args)
+        elif args[0][:2] == ['lxc', 'exec']:
+            return self._lxc_exec(args)
+        elif args[0][:4] == ['lxc', 'image', 'list', '--format=json']:
+            return (
+                '[{"architecture":"test-architecture",'
+                '"fingerprint":"test-fingerprint",'
+                '"created_at":"test-created-at"}]').encode('utf-8')
+        elif args[0][0] == 'sha384sum':
+            return 'deadbeef {}'.format(args[0][1]).encode('utf-8')
+        elif '/usr/lib/sftp-server' in args[0]:
+            return self._popen(args[0])
+        else:
+            return ''.encode('utf-8')
+
     def check_output_side_effect(self):
-        def call_effect(*args, **kwargs):
-            if args[0] == ['lxc', 'remote', 'get-default']:
-                return 'local'.encode('utf-8')
-            elif args[0][:2] == ['lxc', 'info']:
-                return '''
-                    environment:
-                      kernel_architecture: {}
-                    '''.format(self.kernel_arch).encode('utf-8')
-            elif args[0][:3] == ['lxc', 'list', '--format=json']:
-                if self.status and args[0][3] == self.name:
-                    return string.Template('''
-                        [{"name": "$NAME",
-                          "status": "$STATUS",
-                          "devices": $DEVICES}]
-                        ''').substitute({
-                            # Container name without remote prefix
-                            'NAME': self.name.split(':')[-1],
-                            'STATUS': self.status,
-                            'DEVICES': self.devices,
-                            }).encode('utf-8')
-                return '[]'.encode('utf-8')
-            elif (args[0][0] == 'lxc' and
-                  args[0][1] in ['init', 'start', 'launch', 'stop']):
-                return self._lxc_create_start_stop(args)
-            elif args[0][:2] == ['lxc', 'exec']:
-                return self._lxc_exec(args)
-            elif args[0][0] == 'sha384sum':
-                return 'deadbeef {}'.format(args[0][1]).encode('utf-8')
-            elif '/usr/lib/sftp-server' in args[0]:
-                return self._popen(args[0])
-            else:
-                return ''.encode('utf-8')
-        return call_effect
+        return self.call_effect
```

Comment:
Why are you moving this around? Can you clarify?

Reply:
It was failing the complexity check. I tried hard to split the function but failed miserably.
This needs some rethinking, because it's a little hard to read and maintain, and now the next condition we add will fail again.

Reply:
Hrm, leave it for now I guess, any split in there is arbitrary.